### PR TITLE
Refactor: trim wallet state

### PR DESCRIPTION
**Pull request by rocky-fleek**
## Summary
Trims wallet state to not hold assets or collections and thus save up memory and storage space. 

- Removes `collections` from state
- Removes `assets` from state
- Removes unused fields like `customCollections` and `contacts`
- Makes `accountId` a getter.

### CODE REVIEW

File: src/PlugKeyRing/index.ts [hunk 1]
```diff
@@ -53,10 +53,10 @@ class PlugKeyRing {
   // wallet methods
   public getBalances: (args?: { subaccount?: string }) => Promise<Array<TokenBalance>>;
   public getNFTs: (args?: { subaccount?: string, refresh?: boolean }) => Promise<WalletNFTCollection[] | null>;
-  public transferNFT: (args: { subaccount?: string; token: NFTDetails; to: string; standard: string; }) => Promise<WalletNFTCollection[]>;
+  public transferNFT: (args: { subaccount?: string; token: NFTDetails; to: string; standard: string; }) => Promise<boolean>;
   public burnXTC: (args?: { to: string; amount: string; subaccount: string; }) => Promise<TokenInterfaces.BurnResult>;
-  public registerToken: (args: { canisterId: string; standard?: string; subaccount?: string; logo?: string; }) => Promise<Array<TokenBalance>>;
-  public removeToken: (args: { canisterId: string; subaccount?: string; }) => Promise<Array<TokenBalance>>;
+  public registerToken: (args: { canisterId: string; standard?: string; subaccount?: string; logo?: string; }) => Promise<TokenBalance>;
+  public removeToken: (args: { canisterId: string; subaccount?: string; }) => Promise<Array<StandardToken>>;
```
Comment (ignaciosantise) [inline src/PlugKeyRing/index.ts:59]: im thinking about the return value, do we need that? In the app i don't use it

File: src/PlugWallet/index.ts [hunk 2]
```diff
@@ -111,14 +84,9 @@ class PlugWallet {
     this.walletId = walletId;
     this.orderNumber = orderNumber;
     this.walletNumber = walletNumber;
-    this.assets = assets;
     this.icnsData = icnsData;
     this.identity = identity;
-    this.accountId = getAccountId(identity.getPrincipal());;
```
Comment (ignaciosantise) [inline src/PlugWallet/index.ts:117]: is this `accountId` accesible from elsewhere?

File: src/PlugWallet/index.ts [hunk 3]
```diff
@@ -221,16 +193,10 @@ class PlugWallet {
         Principal.fromText(to),
         parseInt(token.index.toString(), 10)
       );
-      // Optimistically update the state
-      const collections = this.collections.map(col => ({
-        ...col,
-        tokens: col.tokens.filter(tok => tok.index !== token.index),
-      }));
-      this.collections = collections.filter(col => col.tokens.length);
       getCachedUserNFTs({ userPID: this.principal, refresh: true }).catch(
```
Comment (ignaciosantise) [inline src/PlugWallet/index.ts:196]: what's `getCachedUserNFTs` used for?

File: src/PlugWallet/index.ts (hunk 2 above)
Comment (tomiir) [reply src/PlugWallet/index.ts:117]: the behavior is the same as I exposed a getter for it which is called exactly the same way. 
And in no other place it's updated so it should be OK!

File: src/PlugWallet/index.ts (hunk 3 above)
Comment (tomiir) [reply src/PlugWallet/index.ts:196]: Requests nfts from kyasshu's cache layer

**Review by ignaciosantise — APPROVED** [review]

**Pull request merged**